packages/core/src/external/sftp/surescripts/schema/shared.ts

```diff
@@ -0,0 +1,250 @@
+// Wraps an SFTP file operation with a schema for each line of the generated file.
+export type FileFieldSchema<T extends object> = {
+  [K in keyof T]: FileField<T, K>;
+}[keyof T][];
+
+export type OutgoingFileRowSchema<T extends object> = {
+  [K in keyof T]: OutgoingFileField<T, K>;
+}[keyof T][];
```

Comment:
This ensures that the array corresponds to every key of type (rather than simply `Array<{ key: keyof T, ... }>`)